For https://github.com/batect/batect/issues/553 (#1301)
batect/batect.dev: docs/getting-started/comparison.md

- Dojo lacks more advanced features that Batect provides to make working with Docker easier and faster, such as
  [cache volumes](../concepts/caches.md) and automatic configuration of [proxies](../how-to/proxies.mdx).

## Batect compared to Earthly

[Earthly](https://earthly.dev/) is focused on bringing CI/CD to your local
development, and is similar to Batect in the goals but differs in particulars
and focus.
Earthly relies on [BuildKit](https://github.com/moby/buildkit) which is a part
of more recent versions of Docker.

Where Earthly is like Batect:

- Both use Docker containers to isolate your build from the local environment
- Both run locally the same build you have in CI (assuming you set up CI to
  use Batect or Earthly)
- Both let you compose your build so everything is not in one file
- Both offer options for parallelization and caching to speed up your builds
  (see below on differences for caching)
- Both let you call out to your existing build, testing, or deployment tooling
- Both have strong documentation

Where Earthly is different from Batect:

- Earthly is an installed program to your system (Batect is self-contained).

>   There are GitHub Actions to add Earthly to your workflow script

Reviewer: ```suggestion
  There are GitHub Actions to add Earthly to your workflow script.
```

- Earthly is both a commercial product, and an open-source project.
  It provides paid-for plans to use their CI infrastructure, however it also

>   works fine as a standalone program in your local and CI environments

Reviewer: ```suggestion
  works fine as a standalone program in your local and CI environments.
```

- Earthly has a larger support community
- Earthly is more focused on Docker, less abstracted from details than Batect

> - Caching has overlapping and dissimilar features.
>   Earthly is closer to Docker-style caching of layers

Reviewer: ```suggestion
- Earthly and Batect have overlapping but also different caching features.
  Earthly is closer to Docker-style caching of layers.
```

- Earthly configuration uses syntax like that of `Dockerfile` (called
  `Earthfile`) with some syntax borrowed from `Makefile` (but without the
  obsession for `<TAB>`s).

>   Batect uses YAML for configuration syntax

Reviewer: ```suggestion
  Batect uses YAML for configuration syntax.
```

> - Earthly and Batect cache for you in different situations.

Reviewer: I'd probably move this point up to near the earlier point about caching features.

>   Earthly tends to be more automatic for caching

Reviewer: ```suggestion
  Earthly tends to be more automatic for caching.
```

> - Earthly is very verbose, and does not have options to quiet the output.
>   (Try the `-o quiet` option in Batect)

Reviewer: ```suggestion
- Earthly is very verbose, and does not have options to quiet the output, whereas Batect has the `--output=quiet` option to show only minimal output.
```

> - Configuration for Earthly can be more specific than Batect, but as a result
>   also be more flexible. Each project needs to evaluate this for themselves

Reviewer: ```suggestion
- Configuration for Earthly can be more specific than Batect, but as a result
  also be more flexible. Each project needs to evaluate this for themselves.
```

  use Batect or Earthly).

- Both let you compose your build so everything is not in one file.

- Both offer options for parallelisation and caching to speed up your builds
  (see below on differences for caching).

- Both let you call out to your existing build, testing, or deployment tooling

- Both have strong documentation.

Where Earthly is different from Batect:

- Earthly is an installed program to your system (Batect is self-contained).
  There are GitHub Actions to add Earthly to your workflow script.

- Earthly is both a commercial product, and an open-source project.
  It provides paid-for plans to use their CI infrastructure, however it also
  works fine as a standalone program in your local and CI environments.

- Earthly has a larger support community.

- Earthly is more focused on Docker, less abstracted from details than Batect.

- Earthly and Batect have overlapping and also different caching features.
  Earthly is closer to Docker-style caching of layers.

- Earthly and Batect cache for you in different situations.
  Earthly tends to be more automatic for caching but also less under your
  control without additional configuration.

> - Earthly configuration uses syntax like that of `Dockerfile` (called
>   `Earthfile`) with some syntax borrowed from `Makefile` (but without the
>   obsession for `<TAB>`s).

Reviewer: ```suggestion
- Earthly configuration uses syntax like that of `Dockerfile` (called
  `Earthfile`) with some syntax borrowed from `Makefile`.
```
